android/src/main/java/io/ably/lib/rest/Push.java

```diff
@@ -0,0 +1,247 @@
+package io.ably.lib.rest;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import io.ably.lib.http.Http;
+import io.ably.lib.http.HttpUtils;
+import io.ably.lib.realtime.CompletionListener;
+import io.ably.lib.types.AblyException;
+import io.ably.lib.types.Callback;
+import io.ably.lib.types.Param;
+import io.ably.lib.types.ErrorInfo;
+import io.ably.lib.util.Log;
+import io.ably.lib.util.Serialisation;
+import android.content.SharedPreferences;
+import android.content.Context;
+import android.content.Intent;
+import android.preference.PreferenceManager;
+import android.support.v4.content.LocalBroadcastManager;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Created by tcard on 3/2/17.
+ */
+public class Push {
+    public Push(AblyRest rest) {
+        this.rest = rest;
+    }
+
+    public void publish(Param[] recipient, JsonObject payload) throws AblyException {
+        rest.http.post("/push/publish", HttpUtils.defaultAcceptHeaders(rest.options.useBinaryProtocol), null, publishBody(recipient, payload), null);
+    }
+
+    public void publishAsync(Param[] recipient, JsonObject payload, final CompletionListener listener) {
+        rest.asyncHttp.post("/push/publish", HttpUtils.defaultAcceptHeaders(rest.options.useBinaryProtocol), null, publishBody(recipient, payload), null, new Callback<Void>() {
+            @Override
+            public void onSuccess(Void result) { listener.onSuccess(); }
+            @Override
+            public void onError(ErrorInfo reason) { listener.onError(reason); }
+        });
+    }
+
+    private Http.RequestBody publishBody(Param[] recipient, JsonObject payload) {
+        JsonObject recipientJson = new JsonObject();
+        for (Param param : recipient) {
+            recipientJson.addProperty(param.key, param.value);
+        }
+        JsonObject bodyJson = new JsonObject();
+        bodyJson.add("recipient", recipientJson);
+        for (Map.Entry<String, JsonElement> entry : payload.entrySet()) {
+            bodyJson.add(entry.getKey(), entry.getValue());
+        }
+        return rest.http.requestBodyFromGson(bodyJson);
+    }
+
+    public void activate(Context context) {
+        SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(context);
+        if (prefs.getBoolean(SharedPrefKeys.ACTIVATED, false)) {
```

Comment:
If you mean if push notifications are enabled on Android's part, Android apps don't need to ask for permissions to publish notifications as iOS apps do. So it's always enabled. The user can't disable them in that sense.

If you mean if the device is registered for push notifications with Ably, then... yeah, there's no way to check that. So you want a way to actively check with Ably's servers if the device is registered?